File: crates/mithril-oxide-sys/cpp/Dialect/MLProgram/IR/MLProgramTypes.cpp
#include "MLProgramTypes.hpp"

#include <mlir/Dialect/MLProgram/IR/MLProgramTypes.h>
#include <mlir/IR/MLIRContext.h>


namespace mithril_oxide_sys::ml_program {

const void *TokenType_get(MLIRContext *ctx)
{
    return TokenType::get(ctx).getAsOpaquePointer();
}

} // mithril_oxide_sys::ml_program

File: crates/mithril-oxide-sys/src/Dialect/MLProgram/IR/MLProgramTypes.rs
pub use self::ffi::TokenType_get;

#[cxx::bridge]
mod ffi {
    #[namespace = "mlir"]
    unsafe extern "C++" {
        include!("mithril-oxide-sys/cpp/Dialect/MLProgram/IR/MLProgram.hpp");

        type MLIRContext = crate::IR::MLIRContext::MLIRContext;
    }

    #[namespace = "mithril_oxide_sys::ml_program"]
    unsafe extern "C++" {
        include!("mithril-oxide-sys/cpp/Dialect/MLProgram/IR/MLProgram.hpp");

        type c_void = crate::IR::Value::ffi::c_void;

        pub unsafe fn TokenType_get(type_: *const c_void) -> *const c_void;
    }
}
